### PR TITLE
NRPT 333, NRPT-374, FLCN-37: Combining mine record add code. Creating universal delete. Fixing news and collection endpoints.

### PR DESCRIPTION
**Pull request by danieltruong**
- https://bcmines.atlassian.net/browse/FLCN-37
- https://bcmines.atlassian.net/browse/NRPT-333
- https://bcmines.atlassian.net/browse/NRPT-374

Things that have changed:
- Adding record within a mine
- Editing record within a mine
- Deleting a record from mine UI
- Deleting news item
- Deleting collection

Note: There is still work required pulling collection and new out from the record controller.

### CODE REVIEW

File: api/src/controllers/delete/delete.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  try {
+    const filter = {
+      _flavourRecords: {
+        $in: [new ObjectID(flavourId)]
+      }
+    };
+
+    switch (flavourType) {
+      case 'bcmi':
+        console.log('wtf');
```
[the remark this answers is not in the document]
Comment (danieltruong) [reply api/src/controllers/delete/delete.js:81]: lol

File: angular/projects/admin-nrpti/src/app/mines/mines-records-edit/mines-records-edit.component.ts [hunk 2]
```diff
@@ -72,22 +69,18 @@ export class MinesRecordsAddEditComponent implements OnInit {
...
-          this.populateTextFields();
-        } else {
-          alert('Error: could not load record.');
-          this.router.navigate(['mines']);
-        }
-      } else {
+      if (res && res.record && res.record[0] && res.record[0].data
+        && res.record[0].data.searchResults && res.record[0].data.searchResults[0]) {
+        this.record = res.record[0].data.searchResults[0];
+        console.log(this.record);
```
Comment (KitArmstrong) [inline angular/projects/admin-nrpti/src/app/mines/mines-records-edit/mines-records-edit.component.ts:75]: Console log.

File: api/src/controllers/collection-controller.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const Delete = require('../controllers/delete/delete');
+let queryActions = require('../utils/query-actions');
```
[the remark this answers is not in the document]
Comment (danieltruong) [reply api/src/controllers/collection-controller.js]: 9ccc241cceafd05e487e51e17ba0490c88fd73e0

File: api/src/controllers/delete/delete.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+const ObjectID = require('mongodb').ObjectID;
+const mongodb = require('../../utils/mongodb');
+const defaultLog = require('../../utils/logger')('record');
+const DocumentController = require('../document-controller')
+
+// This will fully delete a master record.
+// This includes documents and flavours.
+exports.deleteMasterRecord = async function (record) {
+  const db = mongodb.connection.db(process.env.MONGODB_DATABASE || 'nrpti-dev');
+  const collection = db.collection('nrpti');
```
[the remark this answers is not in the document]
Comment (danieltruong) [reply api/src/controllers/delete/delete.js]: 9ccc241cceafd05e487e51e17ba0490c88fd73e0

File: api/src/controllers/delete/delete.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+const ObjectID = require('mongodb').ObjectID;
+const mongodb = require('../../utils/mongodb');
+const defaultLog = require('../../utils/logger')('record');
+const DocumentController = require('../document-controller')
+
+// This will fully delete a master record.
+// This includes documents and flavours.
+exports.deleteMasterRecord = async function (record) {
+  const db = mongodb.connection.db(process.env.MONGODB_DATABASE || 'nrpti-dev');
```
[the remark this answers is not in the document]
Comment (danieltruong) [reply api/src/controllers/delete/delete.js:15]: 9ccc241cceafd05e487e51e17ba0490c88fd73e0

File: api/src/controllers/delete/delete.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (danieltruong) [reply api/src/controllers/delete/delete.js]: 9ccc241cceafd05e487e51e17ba0490c88fd73e0

File: api/src/controllers/news-controller.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+const Delete = require('../controllers/delete/delete');
+let queryActions = require('../utils/query-actions');
```
[the remark this answers is not in the document]
Comment (danieltruong) [reply api/src/controllers/news-controller.js]: 9ccc241cceafd05e487e51e17ba0490c88fd73e0

File: angular/projects/admin-nrpti/src/app/mines/mines-records-edit/mines-records-edit.component.ts (hunk 2 above)
Comment (danieltruong) [reply angular/projects/admin-nrpti/src/app/mines/mines-records-edit/mines-records-edit.component.ts]: 9ccc241cceafd05e487e51e17ba0490c88fd73e0

File: api/src/controllers/delete/delete.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+          }
+        );
+        break;
+      default:
+        defaultLog.info(`protectedDelete - flavour type not supported: ${flavourType}`);
+        throw `protectedDelete - flavour type not supported: ${flavourType}`;
+    }
+    return await this.deleteById(flavourId);
+  } catch (error) {
+    console.log(error);
```
Comment (KitArmstrong) [inline api/src/controllers/delete/delete.js:130]: console log.

File: api/src/controllers/news-controller.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+const queryActions = require('../utils/query-actions');
+const defaultLog = require('../utils/logger')('record');
+
+exports.protectedOptions = function (args, res, next) {
+  res.status(200).send();
+};
+
+exports.protectedDelete = async function (args, res, next) {
+  let newsId = null;
+  if (args.swagger.params.newsId || args.swagger.params.newsId.value) {
```
Comment (KitArmstrong) [inline api/src/controllers/news-controller.js:11]: This should probably be `&&`

File: api/src/controllers/record-controller.js [hunk 9]
```diff
@@ -220,56 +225,51 @@ exports.protectedPut = async function (args, res, next) {
...
-    } catch (e) {
-      defaultLog.info(`protectedNewsDelete - couldn't find record for recordId: ${recordId}`);
-      return queryActions.sendResponse(res, 404, {});
-    }
+exports.protectedDelete = async function (args, res, next) {
+  const db = mongodb.connection.db(process.env.MONGODB_DATABASE || 'nrpti-dev');
+  const collection = db.collection('nrpti');
+  
+  let recordId = null;
+  if (args.swagger.params.recordId || args.swagger.params.recordId.value) {
```
Comment (KitArmstrong) [inline api/src/controllers/record-controller.js:233]: Should probably be `&&`

File: api/src/controllers/collection-controller.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
+const Delete = require('../controllers/delete/delete');
```
[the remark this answers is not in the document]
Comment (danieltruong) [reply api/src/controllers/collection-controller.js:1]: I was going to make a ticket for this. We have the other operations created, but they are currently in the record controller. I will create basic versions of this and later we should migrate the code from the record controller to their correct controllers.

File: api/src/controllers/record-controller.js (hunk 9 above)
Comment (danieltruong) [reply api/src/controllers/record-controller.js]: dd5d034

File: api/src/controllers/news-controller.js (hunk 8 above)
Comment (danieltruong) [reply api/src/controllers/news-controller.js]: dd5d034

File: api/src/controllers/delete/delete.js (hunk 7 above)
Comment (danieltruong) [reply api/src/controllers/delete/delete.js]: dd5d034

File: api/src/controllers/news-controller.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
+const Delete = require('../controllers/delete/delete');
```
[the remark this answers is not in the document]
Comment (danieltruong) [reply api/src/controllers/news-controller.js:4]: dd5d034

File: api/src/controllers/collection-controller.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (danieltruong) [reply api/src/controllers/collection-controller.js:4]: dd5d034

**Pull request merged**